src/java.desktop/share/classes/javax/swing/GroupLayout.java

```diff
@@ -2454,7 +2457,9 @@ public String toString() {
      * @see #createBaselineGroup(boolean,boolean)
      * @since 1.6
      */
-    public class ParallelGroup extends Group {
+    public sealed class ParallelGroup extends Group
+         permits BaselineGroup {
```

Comment:
The BaselineGroup class is private so not part of the public API, is it fine to mention the non-public class in the permits block for the public class?

Reply:
Yes, that is fine; javadoc will filter out displaying non-public classes in a permits clause.

Reply:
Just to double-check, it is fine to have it in the output of the getPermittedSubclasses for the public class as well?

Reply:
I'm not certain what question you are asking. If the question is, is it fine for core reflection to return non-public information about the class, in general sure. For example, in jshell evaluating
StringBuilder.class.getSuperclass()
will yield
 class java.lang.AbstractStringBuilder
which is the non-public superclass of StringBuffer and StringBuilder.

If the question is, do the compatibility expectations of the platform include such visible non-public implementation artifacts? The answer is no; it is fine for those details to evolve and users shouldn't rely on them.

Is there another aspect of the change that was a concern?

Reply:
Thank you for your clarification.

Reply:
You have to include it in the list, else it would not be a permitted sub-class, and then you couldn't use the new syntax at all.

the javadoc doesn't mention it - it still just says
public sealed class GroupLayout.ParallelGroup
extends [GroupLayout.Group]

My recollection is that if there are also public classes the generated javadocs cites the public one and appends the text "and others" ,,  to mean the private ones .. I don't know why it does that but it still doesn't expose the internal class

Reply:
The motivation for that wording is discussed in JEP 409; it indicates the public list of classes is not exhaustive. So for pattern matching (JEP 406), if the pattern matching cases for the switch expressoin are not exhaustive, a default clause on the switch is required. HTH